File: 0102_binary_tree_level_order_traversal/Solution.scala
// LeetCode 0102 - Binary Tree Level Order Traversal
// https://leetcode.com/problems/binary-tree-level-order-traversal/

import scala.collection.mutable

class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
  var value: Int = _value
  var left: TreeNode = _left
  var right: TreeNode = _right
}

object Solution {
  def levelOrder(root: TreeNode): List[List[Int]] = {
    if (root == null) {
      return List.empty
    }

    val result = mutable.ListBuffer[List[Int]]()
    val queue = mutable.Queue[TreeNode](root)

    while (queue.nonEmpty) {
      val size = queue.size
      val level = mutable.ListBuffer[Int]()
      for (_ <- 0 until size) {
        val node = queue.dequeue()
        level += node.value
        if (node.left != null) {
          queue.enqueue(node.left)
        }
        if (node.right != null) {
          queue.enqueue(node.right)
        }
      }
      result += level.toList
    }

    result.toList
  }
}

File: 0102_binary_tree_level_order_traversal/solution.php
<?php
// LeetCode 0102 - Binary Tree Level Order Traversal
// https://leetcode.com/problems/binary-tree-level-order-traversal/

class TreeNode {
    public $val = 0;
    public $left = null;
    public $right = null;
    function __construct($val = 0, $left = null, $right = null) {
        $this->val = $val;
        $this->left = $left;
        $this->right = $right;
    }
}

class Solution {
    /**
     * @param TreeNode $root
     * @return Integer[][]
     */
    function levelOrder($root) {
        if ($root === null) {
            return [];
        }

        $result = [];
        $queue = [$root];

        while (!empty($queue)) {
            $size = count($queue);
            $level = [];
            for ($i = 0; $i < $size; $i++) {
                $node = array_shift($queue);
                $level[] = $node->val;
                if ($node->left !== null) {
                    $queue[] = $node->left;
                }
                if ($node->right !== null) {
                    $queue[] = $node->right;
                }
            }
            $result[] = $level;
        }

        return $result;
    }
}
